src/test/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataFacadeTests.java

```diff
@@ -178,6 +181,12 @@ public void testPutGeoIpData() throws Exception {
                 assertEquals(index, request.indices()[0]);
                 assertEquals(1, request.maxNumSegments());
                 return null;
+            } else if (actionRequest instanceof UpdateSettingsRequest) {
+                UpdateSettingsRequest request = (UpdateSettingsRequest) actionRequest;
+                assertEquals(1, request.indices().length);
+                assertEquals(index, request.indices()[0]);
+                assertEquals("true", request.settings().get("index.blocks.read_only_allow_delete"));
```

Comment:
Do you know why settings are returned as string despite the actual data type? Say for this one we have defined it as Boolean, but expected value is string true", not the boolean `true`.

Reply:
Yes.
```
public String get(String setting) {
        return toString(settings.get(setting));
    }
```

Let me use getAsBoolean instead.